src/main/java/org/jenkinsci/plugins/workflow/cps/SnippetizerLink.java

```diff
@@ -0,0 +1,187 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.cps;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.Functions;
+import hudson.model.RootAction;
+
+import javax.annotation.Nonnull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.jenkinsci.plugins.workflow.cps.Snippetizer.ACTION_URL;
+
+/**
+ * A link that will show up on the side panel of the snippet generator and other similar pages.
+ * Display order is determined by extension ordinal - highest ordinal first.
+ * 
+ * @author Andrew Bayer
+ */
+public abstract class SnippetizerLink implements ExtensionPoint {
+    private static final Logger LOGGER = Logger.getLogger(SnippetizerLink.class.getName());
+
+    /**
+     * Get the URL this link should point to.
+     */
+    @Nonnull
+    public abstract String getUrl();
+
+    /**
+     * Get the actual URL to use in sidepanel.jelly. If {@link #getUrl()} is not absolute, this will prepend "../" to
+     * {@link #getUrl()} to ensure it will work from the snippetizer and other similar pages. Otherwise it will just
+     * return {@link #getUrl()}.
+     */
+    @Nonnull
+    public final String getDisplayUrl() {
+        String u = getUrl();
+        try {
+            if (new URI(u).isAbsolute()) {
+                return u;
+            } else {
+                return "../" + u;
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.log(Level.WARNING, "Failed to parse URL for {0}: {1}", new Object[] {u, e});
+            return "";
+        }
+    }
+
+    /**
+     * Get the icon information for the link.
+     */
+    @Nonnull
+    public String getIcon() {
+        return "icon-help icon-md";
+    }
+
+    /**
+     * Get the display name for the link.
+     */
+    @Nonnull
+    public abstract String getDisplayName();
+
+    /**
+     * Check whether the link should target a new window - this defaults to false;
```

Comment:
I would think this would be the decision of the GUI embedding the link. If from a configuration screen, `target="_blank"`. If from a sidepanel, no.

Reply:
I was trying to ape the behavior currently present in the sidepanel, where the GDSL and external docs link always go to a new window.